File: my-solutions/0053-maximum-subarray/solution.c
int maxSubArray(int* nums, int numsSize) {
    int maxSum = nums[0], currentSum = nums[0];
    
    for (int i = 1; i < numsSize; i++) {
        currentSum = (currentSum > 0) ? currentSum + nums[i] : nums[i];
        if (currentSum > maxSum)
            maxSum = currentSum;
    }
    
    return maxSum;
}

File: my-solutions/0053-maximum-subarray/solution.java
class Solution {
    public int maxSubArray(int[] nums) {
        int cs=0;
        int max=Integer.MIN_VALUE;
        for(int i=0;i<nums.length;i++){
            cs=Math.max(nums[i],nums[i]+cs);
            max=Math.max(cs,max);
        }
        return max;
    }
}
